### PR TITLE
Make sure to check that meta tags should be added before adding them

### PR DESCRIPTION
**Pull request by SammyIsConfused**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]

##### Description of change
#3345 
There was a bug in the template related to adding meta tags - the call to setGoogleMetaTags causes an infinite loop. This is because the function does not check whether the meta tags to be added already exist, and `googleMetaTags` is a dependency, as per eslint's `exhaustive-deps` rule. 

This change just uses lodash's isEqual function compare the DOM's meta tags with the list from the server, and only add tags if these lists differ. 

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: react-front-end/tsrc/mainui/Template.tsx [hunk 1]
```diff
@@ -364,14 +365,16 @@ export const Template = ({
...
-      });
-      setGoogleMetaTags(newMetaTags);
+      if (!isEqual(newMetaTags, googleMetaTags)) {
+        // The meta tags generated on the server side, separated by new line symbols
+        newMetaTags.forEach((newMetaTag) => {
+          head.appendChild(
+            document.createRange().createContextualFragment(newMetaTag)
+          );
+        });
+        setGoogleMetaTags(newMetaTags);
```
[the remark this answers is not in the document]
Comment (SammyIsConfused) [reply react-front-end/tsrc/mainui/Template.tsx:376]: we can't not do that - the eslint exhaustive-deps rule prevents us from updating the state variable without it being a dependency.

[the remark this answers is not in the document]
Comment (PenghaiZhang) [reply react-front-end/tsrc/mainui/Template.tsx:376]: I have checked the logic again.  We want to add tags when server provides tags, and delete tags when server does not. So we need to save the previously added tags in state. 

I also discussed this similar issue with Tom and we both think  a deep equal comparison is right fix.

**Comment by SammyIsConfused** [comment]
Codebuild seems to have failed during compilation of the kaltura module, at first glance. How odd. 
image
I'll restart the build and see if that helps.

**Comment by PenghaiZhang** [comment]
> Codebuild seems to have failed during compilation of the kaltura module, at first glance. How odd.
> image
> I'll restart the build and see if that helps.

Hi Sam, can you get more error details from the build log ? 


Was your local develop branch up-to-date when you created this branch ?

**Comment by SammyIsConfused** [comment]
> > Codebuild seems to have failed during compilation of the kaltura module, at first glance. How odd.
> > image
> > I'll restart the build and see if that helps.
> 
> Hi Sam, can you get more error details from the build log ?
> 
> Was your local develop branch up-to-date when you created this branch ?

That would probably have been it @PenghaiZhang, it was way out.  I've merged it in now, hopefully that fixes it.

**Comment by SammyIsConfused** [comment]
Alright, so now we have an nvm issue, which @PenghaiZhang tells me is present on other branches - not sure how to proceed.

**Comment by edalex-ian** [comment]
You'll need to merge in develop with NVM fix from #3369

**Comment by SammyIsConfused** [comment]
@edalex-ian thanks for that! Merged in and the build seems much happier so far.

**Review by PenghaiZhang — APPROVED** [review]

**Pull request merged**